### PR TITLE
TPSDWAN153: Model for Devices Attachment added

### PR DESCRIPTION
**Pull request by akashg-crest**
- Added Model for devices_attachment Resource
- Added SaveWithFile functionality in Client

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -108,7 +108,7 @@ func GetClient(clientURL, username, password, proxy, proxyCreds string, rateLimi
 }
 
 //MakeRequest generate the HTTP request with required headers
-func (client *Client) MakeRequest(method, path string, body *container.Container, bodyBytes []byte, authenticated bool) (*http.Request, error) {
+func (client *Client) MakeRequest(method, path string, body *container.Container, bodyBytes []byte, bodyBuffer *bytes.Buffer, authenticated bool) (*http.Request, error) {
```
Comment (AshuSoni-crest) [inline client/client.go]: Cannot we make single parameter for bodyBytes and bodyBuffer? If yes, then please do it.

File: client/client.go [hunk 2]
```diff
@@ -219,7 +219,7 @@ func (client *Client) getSessionID() (string, error) {
 	path := "/j_security_check"
 	body := []byte(fmt.Sprintf("j_username=%s&j_password=%s", client.username, client.password))
 
-	req, err := client.MakeRequest(method, path, nil, body, false)
+	req, err := client.MakeRequest(method, path, nil, body, nil, false)
```
Comment (AshuSoni-crest) [inline client/client.go]: Function has more parameters then required

File: client/client.go [hunk 3]
```diff
@@ -249,7 +249,7 @@ func (client *Client) getSessionID() (string, error) {
 func (client *Client) getToken(jsessionID string) (string, error) {
 	method := "GET"
 	path := "/dataservice/client/token"
-	req, err := client.MakeRequest(method, path, nil, nil, false)
+	req, err := client.MakeRequest(method, path, nil, nil, nil, false)
```
Comment (AshuSoni-crest) [inline client/client.go]: Function has more parameters then required

File: client/client.go (hunk 1 above)
Comment (akashg-crest) [reply client/client.go]: Merged Successfully

File: client/client.go (hunk 2 above)
Comment (akashg-crest) [reply client/client.go]: Resolved

File: client/client.go (hunk 3 above)
Comment (akashg-crest) [reply client/client.go]: Resolved

**Review by AshuSoni-crest — APPROVED** [review]

**Pull request merged**